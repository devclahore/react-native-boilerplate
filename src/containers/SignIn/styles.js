/*
  Author: Taimoor Khan
  GitHub: https://github.com/Taimoormk
  Email: taimoor.m.k AT LIVE.COM 
*/

// ########## Import Dependencies Here ##########
import { StyleSheet } from "react-native";

import { STYLE_CONSTANTS, width } from "../../config/styles";

const styles = StyleSheet.create({
  signInContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: STYLE_CONSTANTS.COLORS.GRAY,
  },
  oAuthContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  oAuthHeading: STYLE_CONSTANTS.H1,
  facebookButton: {
    ...STYLE_CONSTANTS.BUTTON_TITLE,
    width: width - 50,
    borderRadius: 7,
    padding: 10
  },
  googleButton: {
    ...STYLE_CONSTANTS.BUTTON_TITLE,
    width: width - 50,
    borderRadius: 7,
    padding: 10
  },
  siginContainer: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center"
  },
  formInput: {
    width: width - 50,
    borderRadius: 7
  },
  spacer: STYLE_CONSTANTS.SPACER
});

export default styles;
