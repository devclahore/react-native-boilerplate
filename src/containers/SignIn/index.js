/*
  Author: Taimoor Khan
  GitHub: https://github.com/Taimoormk
  Email: taimoor.m.k AT LIVE.COM 
*/

// ########## Import Dependencies Here ##########
import React, { Component, Fragment } from "react";
import { func } from "prop-types";
import { View, Text } from "react-native";
import { TextInput, Button } from "react-native-paper";

// ########## Import Components Here ##########
import { STYLE_CONSTANTS } from "../../config/styles";
import styles from "./styles";
import { loginScreenText } from "./helpers";

class SignIn extends Component {
  state = {
    isLoading: false
  };

  render() {
    const { isLoading } = this.state;
    const { showAppAction } = this.props;
    return (
      <Fragment>
        <View style={styles.signInContainer}>
          <Text style={styles.oAuthHeading}>
            {loginScreenText.oAuthHeading}
          </Text>
          <View style={styles.spacer} />
          <Button
            icon={require("../../assets/images/facebook.png")}
            mode="contained"
            loading={isLoading}
            color={STYLE_CONSTANTS.COLORS.FACEBOOK}
            style={styles.facebookButton}
            onPress={() => showAppAction()}
          >
            {loginScreenText.facebookButton}
          </Button>
          <View style={styles.spacer} />
          <Button
            icon={require("../../assets/images/google.png")}
            mode="contained"
            loading={isLoading}
            color={STYLE_CONSTANTS.COLORS.GOOGLE}
            style={styles.googleButton}
            onPress={() => showAppAction()}
          >
            {loginScreenText.googleButton}
          </Button>
        </View>
      </Fragment>
    );
  }
}

SignIn.propTypes = {
  showAppAction: func
};

export default SignIn;
