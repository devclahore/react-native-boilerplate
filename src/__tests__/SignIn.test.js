/* eslint-disable no-undef */
/*
  Author: Taimoor Khan
  GitHub: https://github.com/Taimoormk
  Email: taimoor.m.k AT LIVE.COM 
*/

// ########## Import Dependencies Here ##########
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import configureStore from 'redux-mock-store';

// ########## Import Components Here ##########
import SignIn from '../containers/SignIn';

// ########## Create Initial State ##########
const fakeInitialState = {
  showAppReducer: {
    showApp: true,
  },
};

const mockStore = configureStore();

test('renders SignIn compomenent correctly', () => {
  const renderer = new ShallowRenderer();
  const showApp = false;
  const store = mockStore(fakeInitialState);
  const showAppAction = () => null;
  renderer.render(<SignIn store={store} showApp={showApp} showAppAction={showAppAction} />);
  const wrapper = renderer.getRenderOutput();
  expect(wrapper).toMatchSnapshot();
});
