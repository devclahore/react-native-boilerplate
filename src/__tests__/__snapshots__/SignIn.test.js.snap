// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders SignIn compomenent correctly 1`] = `
<React.Fragment>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#CCCCCC",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "Quicksand-Bold",
          "fontSize": 16,
          "textTransform": "uppercase",
        }
      }
    >
      Continue with Facebook or Google
    </Text>
    <View
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
        }
      }
    />
    <withTheme(Button)
      color="#4267B2"
      icon={
        Object {
          "testUri": "../../../src/assets/images/facebook.png",
        }
      }
      loading={false}
      mode="contained"
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 7,
          "padding": 10,
          "width": 700,
        }
      }
    >
      Continue with Facebook
    </withTheme(Button)>
    <View
      style={
        Object {
          "marginBottom": 10,
          "marginTop": 10,
        }
      }
    />
    <withTheme(Button)
      color="#4285F4"
      icon={
        Object {
          "testUri": "../../../src/assets/images/google.png",
        }
      }
      loading={false}
      mode="contained"
      onPress={[Function]}
      style={
        Object {
          "borderRadius": 7,
          "padding": 10,
          "width": 700,
        }
      }
    >
      Continue with Google
    </withTheme(Button)>
  </View>
</React.Fragment>
`;
